ckanext/datastore/tests/test_search.py

```diff
@@ -857,6 +856,20 @@ def test_works_with_allowed_functions(self):
         )
         helpers.call_action("datastore_search_sql", sql=sql)
 
+    def test_allowed_functions_are_case_insensitive(self):
```

Comment:
need another test for "quoted" function which is case sensitive which should fail for "UpPeR"(author)

can we also update the error message given back to state this some how?

see: https://stackoverflow.com/a/6311188

Reply:
Even if a function is quoted, it will still pass the whitelist if there's a case-insensitive match. I don't think there's a way to determine whether or not the original was quoted at this point in the code, and I don't think it's really necessary to do. A sensitive function should not have the same name, just in a different case, from something on the whitelist.

Reply:
Okay, looks like I was wrong; it's possible to implement entry-quoting with minimal change. I'll see what I can do about unit-testing it.